File: 1380. Lucky Numbers in a Matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        Set<Integer> rows = new HashSet<>();
        for(int i = 0 ; i < matrix.length; i++) {
            int min = Integer.MAX_VALUE;
            for(int j = 0 ; j < matrix[i].length; j++) {
                min = Math.min(matrix[i][j], min);
            }
            rows.add(min);
        }
        for(int i = 0 ; i < matrix[0].length; i++) {
            int max = Integer.MIN_VALUE;
            for(int j = 0; j < matrix.length; j++){
                max = Math.max(max, matrix[j][i]);
            }
            if(rows.contains(max)) {
                return Arrays.asList(max);
            }
        }        
        
        return Arrays.asList();
    }
}

File: 1380. Lucky Numbers in a Matrix.kt
class Solution {
    fun luckyNumbers (matrix: Array<IntArray>): List<Int> {

        val row = mutableListOf<Int>()
        for(x in matrix.indices) {
            var minValue = Int.MAX_VALUE
            for(y in matrix[x].indices) {
               minValue = minOf(minValue, matrix[x][y])
            }
            row.add(minValue)
        }

        for(y in 0 until matrix[0].size) {
            var maxValue = Int.MIN_VALUE
            for (i in 0 until matrix.size) {
                maxValue = maxOf(matrix[i][y], maxValue)
            }
            if(row.contains(maxValue)) {
                return listOf(maxValue)
            }
        }
        return listOf()

    }
}
